extensions/amp-analytics/0.1/events.js

```diff
@@ -1184,22 +1189,35 @@ export class VisibilityTracker extends EventTracker {
   }
 
   /**
+   * @param {string} reportWhenSpec
    * @return {!Promise}
    */
-  createReportReadyPromise_() {
+  createReportReadyPromise_(reportWhenSpec) {
     const viewer = this.root.getViewer();
 
-    if (!viewer.isVisible()) {
-      return Promise.resolve();
-    }
-
-    return new Promise(resolve => {
-      viewer.onVisibilityChanged(() => {
+    switch (reportWhenSpec) {
+      case 'hidden':
         if (!viewer.isVisible()) {
-          resolve();
+          return Promise.resolve();
         }
-      });
-    });
+
+        return new Promise(resolve => {
+          viewer.onVisibilityChanged(() => {
+            if (!viewer.isVisible()) {
+              resolve();
+            }
+          });
+        });
+      case 'endOfFrame':
+        return new Promise(resolve => {
+          // TODO Only resolve when endOfFrame occurs. How to detect this?
```

Comment:
I'm unclear on this. We want to listen for pagehidden and/or unload, correct? Would it be sufficient to just repeat lines 1204-1210? (And probably therefore refactor...)

Reply:
...I recognize that in plain vanilla JavaScript, I would do:
```
document.onunload = () => { resolve(); }
document.onpagehide = () => { resolve(); }
```
...but am a bit unsure of the proper AMP way to do this.

Reply:
Thanks for the tip!
Done.